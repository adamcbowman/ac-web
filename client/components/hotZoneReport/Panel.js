import React from 'react'
import Panel, { INVERSE } from 'components/panel'
import Shim from 'components/Shim'

export default function InversePanel({ children, ...props }) {
    return (
        <Panel {...props} expandable theme={INVERSE}>
            <Shim horizontal>{children}</Shim>
        </Panel>
    )
}
