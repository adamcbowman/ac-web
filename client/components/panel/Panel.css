@import '../../styles/colors.css';

:root {
    --border: 1px solid var(--gray-light);
}

.Container {
    border-top: var(--border);

    &:last-child {
        border-bottom: var(--border);
    }
}

.Expandable {
    & .Header {
        cursor: pointer;

        & svg {
            & path:not([fill='none']) {
                fill: var(--primary);
            }
        }

        @media print {
            & svg {
                display: none;
            }
        }
    }
}

.Inverse {
    & .Header {
        background-color: var(--black);
        color: white;

        & .Expand {
            & path:not([fill='none']) {
                fill: white;
            }
        }
    }
}

.Header {
    position: relative;
    color: var(--primary);
    font-weight: 700;
    line-height: 3em;
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;

    @media print {
        font-size: 1.5em;
    }
}

.Title {
    padding: 0 0.5em;
}

.Expand {
    height: 3em;
    padding: 0 0.5em;
    float: right;

    &:focus {
        outline: none;
    }
}

.Container--Simple {
    composes: Container;
}

.Container--Inverse {
    composes: Container;
    composes: Inverse;
}

.Container--Simple--Expandable {
    composes: Container;
    composes: Expandable;
}

.Container--Inverse--Expandable {
    composes: Container;
    composes: Expandable;
    composes: Inverse;
}
