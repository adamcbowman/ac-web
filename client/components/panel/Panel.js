import React from 'react'
import PropTypes from 'prop-types'
import CSSModules from 'react-css-modules'
import { compose, withState, withHandlers, onlyUpdateForKeys } from 'recompose'
import styles from './Panel.css'
import { Collapse } from '~/components/misc'
import { Expand } from '~/components/button'
import { titleOf } from '~/utils/string'

export const SIMPLE = 'Simple'
export const INVERSE = 'Inverse'

Panel.propTypes = {
    expandable: PropTypes.bool,
    expanded: PropTypes.bool,
    header: PropTypes.string.isRequired,
    theme: PropTypes.oneOf([INVERSE, SIMPLE]),
    children: PropTypes.node.isRequired,
    onHeaderClick: PropTypes.func.isRequired,
}

// Little hack to allow accurate mesuring even when chlidren have
// margins (first and last children)
// http://stackoverflow.com/questions/9770248/div-height-with-child-margin
const STYLE = {
    paddingTop: 1,
    marginTop: -1,
    paddingBottom: 1,
    marginBottom: -1,
}

function Panel({
    expandable = false,
    header,
    theme = SIMPLE,
    expanded = false,
    onHeaderClick,
    children,
}) {
    const styleName = expandable
        ? `Container--${theme}--Expandable`
        : `Container--${theme}`

    return (
        <div styleName={styleName}>
            <header styleName="Header" onClick={onHeaderClick}>
                {expandable &&
                    <Expand styleName="Expand" expanded={expanded} />}
                <span styleName="Title" title={titleOf(header)}>
                    {header}
                </span>
            </header>
            <div styleName="Content">
                {expandable
                    ? <Collapse collapsed={!expanded}>
                          <div style={STYLE}>
                              {children}
                          </div>
                      </Collapse>
                    : children}
            </div>
        </div>
    )
}

export default compose(
    onlyUpdateForKeys(['expanded', 'children', 'header', 'children']),
    withState('expanded', 'setExpanded', props => props.expanded),
    withHandlers({
        onHeaderClick: props => () => {
            if (!props.expandable) {
                return
            }

            props.setExpanded(!props.expanded)
        },
    }),
    CSSModules(styles)
)(Panel)
