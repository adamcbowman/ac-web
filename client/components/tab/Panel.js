import React from 'react'
import PropTypes from 'prop-types'
import {compose, onlyUpdateForKeys} from 'recompose'
import CSSModules from 'react-css-modules'
import styles from './Tab.css'

Panel.propTypes = {
    active: PropTypes.bool,
    children: PropTypes.node,
}

function Panel({children, active = false}) {
    const styleName = active === true ? 'Panel--Active' : 'Panel'

    return (
        <div styleName={styleName} role='tabpanel'>
            {children}
        </div>
    )
}

export default compose(
    onlyUpdateForKeys(['active', 'children']),
    CSSModules(styles),
)(Panel)
