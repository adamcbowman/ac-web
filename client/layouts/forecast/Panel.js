import React from 'react'
import Base, { INVERSE } from 'components/panel'

export default function Panel({ children, ...props }) {
    return (
        <Base {...props} theme={INVERSE} expandable>
            {children}
        </Base>
    )
}
