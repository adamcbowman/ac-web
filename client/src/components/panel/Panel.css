@import "../../styles/colors.css";

:root {
    --border: 1px solid var(--gray-light);
}

.Container {
    border-top: var(--border);

    &:last-child {
        border-bottom: var(--border);
    }
}

.Expandable {
    & .Header {
        cursor: pointer;

        & svg {
            /*transform: translateY(6px);*/
            & path:not([fill="none"]) {
                fill: var(--primary);
            }
        }

        @media print {
            & svg {
                display: none;
            }
        }
    }
}

.Inverse {
    & .Header {
        background-color: var(--black);
        color: white;

        & .Expand {
            float: right;
            & path:not([fill="none"]) {
                fill: white;
            }
        }
    }
}

.Header {
    position: relative;
    color: var(--primary);
    font-weight: 700;
    line-height: 3em;
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;

    @media print {
        font-size: 1.5em;
    }
}

.Title {
    padding: 0 0.5em;
}

.Expand {
    height: 3em;
    padding: 0 0.5em;
    float: left;

    &:focus {
        outline: none;
    }
}


.Content {
    overflow: hidden;
    padding: 0 1em;
    position: relative;

    &:before {
        content: '';
        position: absolute;
        top: 0;
        left: 0;
        right: 0;
        height: 1px;
        background-color: var(--gray-lightest);
    }

    @media print {
        & * {
            height: auto !important;
        }
    }
}

.Container--Simple {
    composes: Container;
}

.Container--Inverse {
    composes: Container;
    composes: Inverse;
}

.Container--Simple--Expandable {
    composes: Container;
    composes: Expandable;
}

.Container--Inverse--Expandable {
    composes: Container;
    composes: Expandable;
    composes: Inverse;
}
