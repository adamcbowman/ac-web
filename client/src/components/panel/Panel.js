import React, {PropTypes} from 'react'
import CSSModules from 'react-css-modules'
import { compose, withState } from 'recompose'
import styles from './Panel.css'
import {Collapse} from 'components/misc'
import {Expand} from 'components/button'

Panel.propTypes = {
    children: PropTypes.arrayOf(PropTypes.node).isRequired,
    expandable: PropTypes.bool,
    expanded: PropTypes.bool,
    header: PropTypes.string.isRequired,
}

// Little hack to allow accurate mesuring even when chlidren have
// margins (first and last children)
// http://stackoverflow.com/questions/9770248/div-height-with-child-margin
const STYLE = {
    paddingTop: 1,
    marginTop: -1,
    paddingBottom: 1,
    marginBottom: -1,
}

function Panel({ expandable = false, header, expanded = false, setExpanded, children }) {
    const styleName = expandable ? 'Header--Expandable' : 'Header'
    function handleClick() {
        if (!expandable) {
            return
        }

        setExpanded(!expanded)
    }

    return (
        <div styleName='Container'>
            <header styleName={styleName} onClick={handleClick}>
                {expandable &&
                    <Expand expanded={expanded} onClick={handleClick} />
                }
                {header}
            </header>
            <div styleName='Content'>
                {expandable ?
                    <Collapse collapsed={!expanded}>
                        <div style={STYLE}>
                            {children}
                        </div>
                    </Collapse>
                : children}
            </div>
        </div>
    )
}

export default compose(
    withState('expanded', 'setExpanded', props => props.expanded)
)(CSSModules(Panel, styles))
