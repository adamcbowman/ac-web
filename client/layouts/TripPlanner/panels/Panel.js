import React, { PureComponent } from 'react'
import styles from '../TripPlanner.css'
import Panel, { INVERSE } from 'components/panel'

export default class TripPlannerPanel extends PureComponent {
    state = {
        expanded: this.props.expanded || false,
    }
    handleExpandedChange = expanded => this.setState({ expanded })
    render() {
        const { children, ...props } = this.props
        const { expanded } = this.state

        return (
            <Panel
                {...props}
                onExpandedChange={this.handleExpandedChange}
                expanded={expanded}
                expandable
                theme={INVERSE}>
                <div className={styles.Panel}>{children}</div>
            </Panel>
        )
    }
}
