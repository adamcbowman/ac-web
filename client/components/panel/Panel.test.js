import React from 'react'
import Renderer from 'react-test-renderer'
import Panel, { SIMPLE, INVERSE } from './Panel'

const THEMES = [SIMPLE, INVERSE]
const header = 'Header'

test('panel component', () => {
    THEMES.forEach(theme => {
        const panel = Renderer.create(
            <Panel header={header} theme={theme}>Content</Panel>
        )

        expect(panel).toMatchSnapshot()
    })

    THEMES.forEach(theme => {
        const panel = Renderer.create(
            <Panel header={header} expanded theme={theme}>Content</Panel>
        )

        expect(panel).toMatchSnapshot()
    })

    THEMES.forEach(theme => {
        const panel = Renderer.create(
            <Panel header={header} expandable theme={theme}>Content</Panel>
        )

        expect(panel).toMatchSnapshot()
    })
})
