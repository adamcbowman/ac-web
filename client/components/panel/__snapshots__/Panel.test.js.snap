// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel component 1`] = `
<section
  className="Container--Simple"
>
  <header
    className="Header"
    onClick={null}
  >
    <span
      className="Title"
    >
      Header
    </span>
  </header>
  <div
    className="Container"
  >
    Content
  </div>
</section>
`;

exports[`panel component 2`] = `
<section
  className="Container--Inverse"
>
  <header
    className="Header"
    onClick={null}
  >
    <span
      className="Title"
    >
      Header
    </span>
  </header>
  <div
    className="Container"
  >
    Content
  </div>
</section>
`;

exports[`panel component 3`] = `
<section
  className="Container--Simple"
>
  <header
    className="Header"
    onClick={null}
  >
    <span
      className="Title"
    >
      Header
    </span>
  </header>
  <div
    className="Container"
  >
    Content
  </div>
</section>
`;

exports[`panel component 4`] = `
<section
  className="Container--Inverse"
>
  <header
    className="Header"
    onClick={null}
  >
    <span
      className="Title"
    >
      Header
    </span>
  </header>
  <div
    className="Container"
  >
    Content
  </div>
</section>
`;

exports[`panel component 5`] = `
<section
  className="Container--Simple--Expandable"
>
  <header
    className="Header"
    onClick={[Function]}
  >
    <button
      className="Expand Subtile"
    >
      <svg
        fill="none"
        height={24}
        viewBox="0 0 24 24"
        width={24}
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          fill="#245EAC"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
    <span
      className="Title"
    >
      Header
    </span>
  </header>
  <div
    className="Container"
  >
    Content
  </div>
</section>
`;

exports[`panel component 6`] = `
<section
  className="Container--Inverse--Expandable"
>
  <header
    className="Header"
    onClick={[Function]}
  >
    <button
      className="Expand Subtile"
    >
      <svg
        fill="none"
        height={24}
        viewBox="0 0 24 24"
        width={24}
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          fill="#245EAC"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
    <span
      className="Title"
    >
      Header
    </span>
  </header>
  <div
    className="Container"
  >
    Content
  </div>
</section>
`;
